### PR TITLE
[INLONG-10704][Sort] Sort Add Oceanbase Support

### LINKED ISSUE #10632: [Feature][Dashboard] dashboard should support oceanbase
### Description

oceanbase is a high-performance distributed database, I hope inlong can support it

### Use case

oceanbase is a high-performance distributed database, I hope inlong can support it

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #10704: [Feature][Sort] sort should support oceanbase
### Description

sort should support oceanbase 

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xxsc0529**
Fixes #10704

### Motivation
Added support for Oceanbase data sources to adapt to the Oceanbase ecosystem. close #10704

### Modifications
It mainly adds oceanbase data source support and uses the oceanbase-client driver to connect.

It mainly adds oceanbase data source support and uses the oceanbase-client driver to connect. And write oceanbase's ExtractNode logic and LoadNode logic. The specific implementation logic is the same as mysql.Added support for oceanbase on the front-end page. Since the jdbc-inlong processing logic in the flink15 connection lacks the jdbc dialect, an error is reported. This is modified here.
### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*
Here's how the page test works
image
image
image
image

- [ ] This change added tests and can be verified as follows:
Please use the following test classes for testing  org.apache.inlong.sort.parser.MySqlExtractNodeToOceanBaseLoadNodeTest
  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no) yes
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)  Data source types have been added to the README.md
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/OceanBaseExtractNode.java [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+import java.io.Serializable;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Mysql extract node using debezium engine
```
Comment (vernedeng) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/OceanBaseExtractNode.java:50]: OceanBase extract node

Comment (xxsc0529) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/OceanBaseExtractNode.java]: Revised

**Review by vernedeng — DISMISSED** [review]

File: inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/MySqlExtractNodeToOceanBaseLoadNodeTest.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+        final List<FieldRelation> fieldRelations = Arrays
+                .asList(new FieldRelation(new FieldInfo("id", new IntFormatInfo()),
+                        new FieldInfo("id", new IntFormatInfo())),
+                        new FieldRelation(new FieldInfo("user_name", new StringFormatInfo()),
+                                new FieldInfo("user_name", new StringFormatInfo())),
+                        new FieldRelation(new FieldInfo("phone", new IntFormatInfo()),
+                                new FieldInfo("phone", new IntFormatInfo())));
+
+        // Support delete event (sink.enable-delete='true'), requires Doris table to enable batch delete function
```
Comment (EMsnap) [inline inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/MySqlExtractNodeToOceanBaseLoadNodeTest.java:83]: doris table ?

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialectFactory.java
Comment (XiaoYou201) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialectFactory.java:1]: This factory should named OceanBaseDialectFactory?

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialectFactory.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package org.apache.inlong.sort.jdbc.dialect.jdbc;
+
+public class JdbcDialectFactory implements org.apache.inlong.sort.jdbc.dialect.JdbcDialectFactory {
+
+    @Override
+    public boolean acceptsURL(String url) {
+        return url.startsWith("jdbc:mysql:");
+    }
```
Comment (XiaoYou201) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialectFactory.java:25]: It should be return url.startsWith("jdbc:oceanbase:")?

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+/**
+ * JDBC dialect for ClickHouse SQL.
+ */
+public class JdbcDialect extends AbstractDialect {
```
Comment (XiaoYou201) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java:42]: This is copy of clickhouse dialect? plz test that the program is working properly.

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+    // Define MAX/MIN precision of DECIMAL type according to ClickHouse docs:
+    // https://clickhouse.com/docs/zh/sql-reference/data-types/decimal/
+    private static final int MAX_DECIMAL_PRECISION = 128;
+    private static final int MIN_DECIMAL_PRECISION = 32;
+    private static final String POINT = ".";
+
+    @Override
+    public String dialectName() {
+        return "mysql";
+    }
```
Comment (XiaoYou201) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java:60]: dialectName is mysql？the url of oceanbase  should be "jdbc:oceanbase" through jdbc protocol

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+    @Override
+    public JdbcRowConverter getRowConverter(RowType rowType) {
+        return new ClickHouseRowConverter(rowType);
+    }
```
Comment (XiaoYou201) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java:65]: clickhouse?

**Comment by xxsc0529** [comment]
> please add test result with real data from any source to oceanbase

What kind of test data is needed and where to put it

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java (hunk 6 above)
Comment (xxsc0529) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java]: has been modified，In the test, it was found that flink needs the dialect to find the corresponding factory, and this code does not affect it, but it has been modified

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialectFactory.java (hunk 3 above)
Comment (xxsc0529) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialectFactory.java:25]: The transformation was done here, Oceanbase officially expects to use MySQL CDC, and the MySQL protocol needs to be used in Flink, OceanBase supports the MySQL protocol, so this becomes the beginning of MySQL

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java (hunk 5 above)
Comment (xxsc0529) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java:55]: The transformation was done here, Oceanbase officially expects to use MySQL CDC, and the MySQL protocol needs to be used in Flink, OceanBase supports the MySQL protocol, so this becomes the beginning of MySQL

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java (hunk 4 above)
Comment (xxsc0529) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java:41]: Yes, it's normal to have tested it, and Flink needs a mysq dialect to find the corresponding factory

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialectFactory.java
Comment (xxsc0529) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialectFactory.java:1]: No, it was found in the test that MySQL and Oceanbase both need this dialect to find the corresponding factory that conforms to the JDBC protocol

File: inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/MySqlExtractNodeToOceanBaseLoadNodeTest.java (hunk 2 above)
Comment (xxsc0529) [reply inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/MySqlExtractNodeToOceanBaseLoadNodeTest.java]: has been modified

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java (hunk 4 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java:41]: Plz modify the comment~ "JDBC dialect for MYSQL" will be better?

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java (hunk 5 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java:55]: LGTM~

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java (hunk 6 above)
Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java]: LGTM~

**Comment by XiaoYou201** [comment]
> > please add test result with real data from any source to oceanbase
> 
> What kind of test data is needed and where to put it

You can refer to inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/Mysql2StarRocksTest.java

or package your code to manager container. Then start a job to verify the oceanbase connector work properly.If all right, plz 
provide some screenshots~

**Comment by xxsc0529** [comment]
> > > please add test result with real data from any source to oceanbase
> > 
> > 
> > What kind of test data is needed and where to put it
> 
> You can refer to inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/Mysql2StarRocksTest.java
> 
> or package your code to manager container. Then start a job to verify the oceanbase connector work properly.If all right, plz provide some screenshots~

 The logic of Oceanbase and MySQL is the same, using MySQL-CDC, and the following is tested in the form of a remote debug project jar package through local source code
image
This step shows that the pre-configuration logic is passable, and the corresponding CDC can be found and flinksql can be generated，Below is the generated flinksql
FlinkSqlParseResult(tableEnv=org.apache.flink.table.api.bridge.java.internal.StreamTableEnvironmentImpl@42554211, createTableSqls=[CREATE TABLE `table_mysqtooceanbase`(
    `id` INT,
    `user_name` STRING)
    WITH (
    'inlong.metric.labels' = 'groupId=mysqtooceanbase&streamId=mysqtooceanbase&nodeId=mysqtooceanbase',
    'topic' = 'mysqtooceanbase.mysqtooceanbase',
    'properties.bootstrap.servers' = '127.0.0.1:9092',
    'connector' = 'kafka-inlong',
    'inlong-msg.debezium-json.schema-include' = 'false',
    'inlong-msg.debezium-json.encode.decimal-as-plain-number' = 'true',
    'inlong-msg.inner.format' = 'debezium-json',
    'format' = 'inlong-msg',
    'inlong-msg.debezium-json.timestamp-format.standard' = 'SQL',
    'inlong-msg.debezium-json.ignore-parse-errors' = 'true',
    'inlong-msg.debezium-json.map-null-key.literal' = 'null',
    'inlong-msg.debezium-json.map-null-key.mode' = 'DROP',
    'inlong-msg.ignore-parse-errors' = 'false',
    'scan.startup.mode' = 'earliest-offset',
    'properties.group.id' = 'kafka-local_mysqtooceanbase.mysqtooceanbase_consumer_group'
), CREATE TABLE `table_oceanbase`(
    PRIMARY KEY (`id`) NOT ENFORCED,
    `id` INT,
    `user_name` STRING)
    WITH (
    'inlong.metric.labels' = 'groupId=mysqtooceanbase&streamId=mysqtooceanbase&nodeId=oceanbase',
    'connector' = 'jdbc-inlong',
    'url' = 'jdbc:mysql://Confidentiality:4848/test',
    'username' = 'Confidentiality@tt1',
    'password' = '123456',
    'table-name' = 't_ds_user'
)], loadSqls=[INSERT INTO `table_oceanbase`
    SELECT 
    `id` AS `id`,
    `user_name` AS `user_name`
    FROM `table_mysqtooceanbase` ])
Go to flink-ui to find that the task has also been submitted
image

**Comment by XiaoYou201** [comment]
LGTM~

**Review by XiaoYou201 — APPROVED** [review]

**Review by vernedeng — DISMISSED** [review]

**Review by aloyszhang — DISMISSED** [review]

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+/**
+ * JDBC dialect for ClickHouse SQL.
```
Comment (EMsnap) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java:39]: JDBC for mysql SQL?

Comment (xxsc0529) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/jdbc/JdbcDialect.java]: has been modified

**Review by dockerzhang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**